### PR TITLE
add(types): nsfw_level field to Guild

### PR DESCRIPTION
**Pull request by rigormorrtiss**
Reference: https://github.com/discord/discord-api-docs/pull/2976

### CODE REVIEW

File: src/types/guilds/guild_nsfw_level.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// TODO: add resource link
+export enum GuildNSFWLevel {
```
Comment (Skillz4Killz) [inline src/types/guilds/guild_nsfw_level.ts:2]: I would make this GuildNsfwLevel to just fit the camelCase theme

File: .gitignore [hunk 2]
```diff
@@ -26,3 +26,6 @@ desktop.ini
 !.vscode/tasks.json
 !.vscode/launch.json
 *.code-workspace
+
+# Sublime Text
+.sublime-project
```
Comment (Skillz4Killz) [inline .gitignore:31]: Remove everything in this file besides debug.ts

File: src/types/guilds/guild.ts [hunk 3]
```diff
@@ -111,8 +112,8 @@ export interface Guild {
   approximatePresenceCount?: number;
   /** The welcome screen of a Community guild, shown to new members, returned in an Invite's guild object */
   welcomeScreen?: WelcomeScreen;
-  /** `true` if this guild is designated as NSFW */
-  nsfw: boolean;
+  /** Guild NSFW level */
+  nsfwLevel: GuildNSFWLevel;
```
Comment (rigormorrtiss) [inline src/types/guilds/guild.ts:116]:
```suggestion
  /** Guild NSFW level */
  nsfwLevel: GuildNsfwLevel;
```

File: src/types/guilds/guild_nsfw_level.ts (hunk 1 above)
Comment (rigormorrtiss) [inline src/types/guilds/guild_nsfw_level.ts:2]:
```suggestion
export enum GuildNsfwLevel {
```

File: src/types/guilds/guild.ts [hunk 4]
```diff
@@ -13,6 +13,7 @@ import { DiscordSystemChannelFlags } from "./system_channel_flags.ts";
 import { DiscordVerificationLevels } from "./verification_levels.ts";
 import { WelcomeScreen } from "./welcome_screen.ts";
 import type { StageInstance } from "../channels/stage_instance.ts";
+import { GuildNSFWLevel } from "./guild_nsfw_level.ts";
```
Comment (itohatweb) [inline src/types/guilds/guild.ts:16]:
```suggestion
import { GuildNsfwLevel } from "./guild_nsfw_level.ts";
```

**Review by itohatweb — CHANGES_REQUESTED** [review]

**Review by itohatweb — APPROVED** [review]

**Pull request merged**